Marlin/src/feature/mmu2/mmu2.cpp

```diff
@@ -667,8 +781,9 @@ void MMU2::filament_runout() {
 }
 
 #if ENABLED(PRUSA_MMU2_S_MODE)
+
   void MMU2::check_filament() {
-    const bool runout = READ(FIL_RUNOUT_PIN) ^ (FIL_RUNOUT_INVERTING);
+    const bool runout = FILAMENT_PRESENT();
```

Comment:
@BastR — This looks like it should be `!FILAMENT_PRESENT()` instead, but so far no one has mentioned a problem with the existing code. What do you think?

Otherwise, it seems ready to merge.

Reply:
This part of the code is not from me, it comes from the previous PR https://github.com/MarlinFirmware/Marlin/pull/17523. 
But I think the term "runout" is misused, in this function we check that the filament is present and if it is, we send the command "A" to the MMU to stop loading. 
So it would be more correct to use something like that:
```
void MMU2::check_filament() {
const bool filament_reached_gears = FILAMENT_PRESENT();
```

I choose "filament reached gears" because this part of the code work only if the printer have an MK3S style extruder.

Reply:
Please give me one more day to run some tests and I'll tag you so you can merge.

Reply:
I look forward to hearing that it went well….